core/shared/src/main/scala/zio/ZSchedule.scala

```diff
@@ -378,82 +395,108 @@ trait ZSchedule[-R, -A, +B] extends Serializable { self =>
     new ZSchedule[R, A1, C] {
       type State = self.State
       val initial = self.initial
-      val update  = (a: A1, s: State) => self.update(a, s).map(_.rightMap(f))
+      val extract = (a: A1, s: self.State) => f(self.extract(a, s))
+      val update  = self.update
     }
 
   /**
    * Returns a new schedule with the specified effectful modification
-   * applied to each delay produced by this schedule.
+   * applied to each sleep performed by this schedule.
+   *
+   * Note that this does not apply to sleeps performed in ZSchedule#initial.
```

Comment:
Is there a reason to exclude that? For backward compatibility?

Reply:
We need the current B to decide how to modify sleeps, but sadly cant extract it for initial